diff --git a/scripts/build_fm_images.sh b/scripts/build_fm_images.sh
index de120d9..0c4246e 100755
--- a/scripts/build_fm_images.sh
+++ b/scripts/build_fm_images.sh
@@ -63,7 +63,7 @@ build_final_images()
    # set fm version in docker_compose file
    mkdir -p static
    cp misc/docker_compose_untagged.yml static/docker_compose_v$FM_VERSION.yml 
-   sed -i "s/fm_version/$FM_VERSION/g" static/docker_compose_v$FM_VERSION.yml
+   sed -i.bak "s/fm_version/$FM_VERSION/g" static/docker_compose_v$FM_VERSION.yml
 }
 
 
